python/ray/data/preprocessors/batch_mapper.py

```diff
@@ -43,9 +44,28 @@ class BatchMapper(Preprocessor):
 
     _is_fittable = False
 
-    def __init__(self, fn: Callable[["pandas.DataFrame"], "pandas.DataFrame"]):
+    def __init__(
+        self,
+        fn: Callable[
+            [Union["pandas.DataFrame", np.ndarray, Dict[str, np.ndarray]]],
+            Union["pandas.DataFrame", np.ndarray, Dict[str, np.ndarray]],
+        ],
+        batch_format="pandas",
```

Comment:
should this default to None?

Reply:
Basically default to the native dataset format unless explicitly overridden by the user.

Reply:
I think our current behavior is already `pandas` for all places we use `BatchMapper` so defaulting to this makes next step migration easier.